### ISSUE TITLE
Bugs report for VEX-encoded-instructions

### ISSUE THREAD

**Issue by adahsuzixin** [lifting-bits/remill#440]
Hi team

I found some bugs in `VEX`-encoded-instructions.
For example, If we add a testcase in [PSHUFD.S](https://github.com/lifting-bits/remill/blob/master/tests/X86/SSE/PSHUFD.S).
```cpp
TEST_BEGIN(VPSHUFDv128v128i8_cf, 1)
TEST_INPUTS(
    0)

    vpshufd xmm4, xmm0, 0xcc
    vpshufd xmm5, xmm1, 0xdd
    vpshufd xmm6, xmm2, 0xee
    vpshufd xmm7, xmm3, 0xff
TEST_END
```

It will fail.

We can find the slightly difference between `PSHUFD` and `VPSHUFD`, As described in [PSHUFD — Shuffle Packed Doublewords](https://www.felixcloutier.com/x86/pshufd):
```
PSHUFD (128-bit Legacy SSE version)
DEST[31:0]←(SRC >> (ORDER[1:0] * 32))[31:0];
DEST[63:32]←(SRC >> (ORDER[3:2] * 32))[31:0];
DEST[95:64]←(SRC >> (ORDER[5:4] * 32))[31:0];
DEST[127:96]←(SRC >> (ORDER[7:6] * 32))[31:0];
DEST[MAXVL-1:128] (Unmodified)

VPSHUFD (VEX.128 encoded version)
DEST[31:0]←(SRC >> (ORDER[1:0] * 32))[31:0];
DEST[63:32]←(SRC >> (ORDER[3:2] * 32))[31:0];
DEST[95:64]←(SRC >> (ORDER[5:4] * 32))[31:0];
DEST[127:96]←(SRC >> (ORDER[7:6] * 32))[31:0];
DEST[MAXVL-1:128] ← 0
```
So it's different between `DEST[MAXVL-1:128] (Unmodified)` and `DEST[MAXVL-1:128] ← 0`.
However, we used the same template in [SSE.cpp](https://github.com/lifting-bits/remill/blob/master/remill/Arch/X86/Semantics/SSE.cpp).
```cpp
DEF_ISEL(PSHUFD_XMMdq_MEMdq_IMMb) = PSHUFD<V128W, MV128>;
DEF_ISEL(PSHUFD_XMMdq_XMMdq_IMMb) = PSHUFD<V128W, V128>;
IF_AVX(DEF_ISEL(VPSHUFD_XMMdq_MEMdq_IMMb) = PSHUFD<V128W, MV128>;)
IF_AVX(DEF_ISEL(VPSHUFD_XMMdq_XMMdq_IMMb) = PSHUFD<V128W, V128>;)
```
It caused problems.

**Comment by pgoodman** [comment]
It looks like what is needed is:
```
IF_AVX(DEF_ISEL(VPSHUFD_XMMdq_MEMdq_IMMb) = PSHUFD<VV128W, MV128>;)
IF_AVX(DEF_ISEL(VPSHUFD_XMMdq_XMMdq_IMMb) = PSHUFD<VV128W, V128>;)
```

Can you test this can submit a PR with a fix?

**Comment by adahsuzixin** [comment]
Yes, this was proved to be right, see [pull/442](https://github.com/lifting-bits/remill/pull/442)

**Comment by pgoodman** [comment]
Fixed in #442.
